src/Microsoft.DotNet.ProjectJsonMigration/MigrationBackupPlan.cs

```diff
@@ -83,8 +85,22 @@ public void PerformBackup()
             {
                 file.MoveTo(
                     Path.Combine(
-                        ProjectBackupDirectory.FullName, file.Name));
+                        ProjectBackupDirectory.FullName,
+                        file.Name));
+            }
+        }
+
+        private static string GetUniqueDirectoryPath(string directoryPath)
+        {
+            var candidatePath = directoryPath;
+
+            var suffix = 1;
+            while (Directory.Exists(directoryPath))
+            {
+                candidatePath = $"{directoryPath}_{suffix++}";
```

Comment:
I have not looked into the VS behavior. Do you know of a good way to produce a project that actually migrates in VS? Since we adding round-tripping support years ago, it's been harder to find projects that it actually forces to migrate. :smile:

IIRC, VS provided a way for the user to specify a backup folder location and then errored if it couldn't copy files there. It seems that https://msdn.microsoft.com/en-us/library/60z6y467(v=vs.90).aspx concurs with my memory somewhat. I'm not sure it's desirable to match that experience, but if you prefer, I could provide a meaningful error message in the CLI instead of trying to create a unique backup folder name. However, that would mean that backup would always fail for projects that happen to have a folder called "backup" in them, which is not unreasonable.